.github/workflows/check_requirements.yml

```diff
@@ -0,0 +1,51 @@
+name: Check the dependencies in requirements.in and requirements.txt
+
+on:
+  workflow_call:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request_target:
+    branches:
+      - main
+
+concurrency:
+  group: check-requirements-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  tests:
+    name: Run requirements tests on multiple architectures
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, ubuntu-20.04]
```

Comment:
why do we want to run on two different os distro?

Reply:
No we don't need it anymore. I will also removing running on two architectures as this is not needed either.